src/Migrations/Version20250325123754.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * MAT-410 – Prepare schema for new allocation order method
+ */
+final class Version20250325123754 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Prepare schema for new allocation order method';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('DROP INDEX available_fundings ON CampaignFunding');
+        $this->addSql('CREATE INDEX available_fundings ON CampaignFunding (amountAvailable, id)');
+        $this->addSql('ALTER TABLE Fund ADD allocationOrder INT NOT NULL');
+        $this->addSql('CREATE INDEX allocationOrder ON Fund (allocationOrder)');
```

Comment:
not tested but I think if you add this you don't need the expand/contract pattern:

```suggestion
        $this->addSql('CREATE INDEX allocationOrder ON Fund (allocationOrder)');
        
        $this->addSql(<<<'SQL'
            UPDATE Fund SET allocationOrder = CASE
                WHEN fundType = 'pledge' THEN 100
                WHEN fundType = 'championFund' THEN 200
                WHEN fundType = 'TopupPledge' THEN 300
            END
        SQL);
```

Reply:
Thanks, yep I think you're right. Sorry I forgot this suggestion and tried to do it with a join in a new PR at https://github.com/thebiggive/matchbot/pull/1283/files#diff-f86833cd06ef4cd40ca911051395ea608ed85db2d86cd250c979872f91d78348R22-R28

I'm not opposed to keeping it all on `Fund` although I think the case above is not right for 'topupPledge'. Not sure which way is safer really, it's hard to test usefully and also do it all on one PR.